plugins/source/azure/client/services/front_door.go

```diff
@@ -8,12 +8,16 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 )
 
-type FrontDoorClient interface {
+type FrontDoorClient struct {
+	Doors FrontDoorDoorsClient
```

Comment:
The naming convention sometimes creates redundancy but I kept it to simplify code generation